tests/util/test_json.py

```diff
@@ -99,3 +105,20 @@ def default(self, o):
     save_json(fname, Mock(), encoder=MockJSONEncoder)
     data = load_json(fname)
     assert data == "9"
+
+
+def test_find_unserializable_data():
+    """Find unserializeable data."""
+    assert find_paths_unserializable_data(1) == []
+    assert find_paths_unserializable_data([1, 2]) == []
+    assert find_paths_unserializable_data({"something": "yo"}) == []
+
+    assert find_paths_unserializable_data({"something": set()}) == ["$.something"]
+    assert find_paths_unserializable_data({"something": [1, set()]}) == [
+        "$.something[1]"
+    ]
+    assert find_paths_unserializable_data([1, {"bla": set(), "blub": set()}]) == [
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5020228)